zebra-chain/src/orchard/action.rs

```diff
@@ -59,13 +59,47 @@ impl ZcashSerialize for Action {
 
 impl ZcashDeserialize for Action {
     fn zcash_deserialize<R: io::Read>(mut reader: R) -> Result<Self, SerializationError> {
+        // # Consensus
+        //
+        // > Elements of an Action description MUST be canonical encodings of the types given above.
+        //
+        // https://zips.z.cash/protocol/protocol.pdf#actiondesc
+        //
+        // > LEOS2IP_{256}(cmx) MUST be less than 𝑞_ℙ.
+        //
+        // https://zips.z.cash/protocol/protocol.pdf#actionencodingandconsensus
+        //
+        // See comments below for each specific type.
         Ok(Action {
+            // Type is ValueCommit^{Orchard}.Output, i.e. ℙ.
+            // https://zips.z.cash/protocol/protocol.pdf#abstractcommit
+            // See [`ValueCommitment::zcash_deserialize`].
             cv: ValueCommitment::zcash_deserialize(&mut reader)?,
+            // Type is `{0 .. 𝑞_ℙ − 1}`. See [`Nullifier::try_from`].
             nullifier: Nullifier::try_from(reader.read_32_bytes()?)?,
+            // Type is SpendAuthSig^{Orchard}.Public, i.e. ℙ.
+            // https://zips.z.cash/protocol/protocol.pdf#concretespendauthsig
+            // https://zips.z.cash/protocol/protocol.pdf#concretereddsa
+            // This only reads the 32-byte buffer. The type is enforced
+            // on signature verification; see zebra-chain/src/primitives/redpallas/batch.rs
```

Comment:
Wouldn't a reference like this work?
```suggestion
            // on signature verification; see [`redpallas::batch`]
```